nncf/data/dataset.py

```diff
@@ -53,6 +53,13 @@ def __init__(self,
         self._data_source = data_source
         self._transform_func = transform_func
 
+    @property
+    def data_source(self):
+        """
+        :return: The iterable object where data items are from.
+        """
+        return self._data_source
+
```

Comment:
Could you please explain why it was added?

Reply:
data_source is used to detect the batch size.

Proofs:
https://github.com/openvinotoolkit/nncf/blob/732e630259bd2fd5b047d180aec40f2d805faf36/nncf/torch/quantization/quantize.py#L48
https://github.com/openvinotoolkit/nncf/blob/732e630259bd2fd5b047d180aec40f2d805faf36/nncf/tensorflow/quantization/quantize.py#L42

Reply:
I got it. Is it possible to determine the batch size by the first element of the dataset?

Reply:
`data_source` should be more precisely described in the docstring. Is it a dataloader, or a dataset? Is there any suppositions about the type or structure of each returned iterable of the `data_source`?

Reply:
Guys, thanks for comment. The data_source property was removed from the current PR because propose to introduce batch_size property in the next PR.